client/src/utils.ts

```diff
@@ -71,12 +66,10 @@ export function switchTheme(theme: string, set: (theme: string) => void) {
 }
 
 export function isPayingSubscriber(user) {
-  const validSubscriptionTypes = [
-    MDN_PLUS_5M_PLAN_NAME,
-    MDN_PLUS_5Y_PLAN_NAME,
-    MDN_PLUS_10M_PLAN_NAME,
-    MDN_PLUS_10Y_PLAN_NAME,
-  ];
+  const validSubscriptionTypes = Object.entries(SubscriptionType)
+    .filter((type) => type[1] !== SubscriptionType.MDN_CORE)
+    .map((type) => type[1]);
+  console.log(validSubscriptionTypes);
```

Comment:
I believe the best would be to split `SubscriptionType` into two `enum`s and combine them using a union:

```ts
export enum FreeSubscriptionType {
  MDN_CORE = "",
}

export enum PlusSubscriptionType {
  MDN_PLUS_5M = "mdn_plus_5m",
  MDN_PLUS_5Y = "mdn_plus_5y",
  MDN_PLUS_10M = "mdn_plus_10m",
  MDN_PLUS_10Y = "mdn_plus_10y",
}

export type SubscriptionType = FreeSubscriptionType | PlusSubscriptionType
```

Then you should be able to get the subscription type values here as follows:

```ts
const validSubscriptionTypes = Object.values(PlusSubscriptionType);
```

Reply:
That seems to break other code that depends on the combined `enum`

Reply:
Specifically `client/src/plus/offer-overview/offer-overview-subscribe/index.tsx`

Reply:
It expects `SubscriptionType` to be an `enum` not a `type`

Reply:
I refactored this a little bit and it now needs one less loop :) I reckon this is good enough. It also avoids the need to touch other pieces of code.